src/lib/storage/lz4_segment.cpp

```diff
@@ -0,0 +1,147 @@
+#include "lz4_segment.hpp"
+
+#include <lz4.h>
+
+#include "resolve_type.hpp"
+#include "storage/vector_compression/base_compressed_vector.hpp"
+#include "utils/assert.hpp"
+#include "utils/performance_warning.hpp"
+
+namespace opossum {
+
+template <typename T>
+LZ4Segment<T>::LZ4Segment(pmr_vector<char> compressed_data, pmr_vector<bool> null_values,
+                          const std::shared_ptr<const pmr_vector<size_t>>& offsets, const size_t decompressed_size)
```

Comment:
Why is offsets a shared_ptr again? Who is it shared with? If it is just because it is only used for strings, `optional` would be the better fit.

Reply:
No other reason than it being easy to use when not set. We'll change it to `optional`